model/src/main/scala/ch.epfl.scala.index.model/release/LanguageVersion.scala

```diff
@@ -37,10 +37,10 @@ final case class ScalaVersion(version: BinaryVersion) extends LanguageVersion {
 final case class Scala3Version(version: BinaryVersion) extends LanguageVersion {
   def family = "scala3"
   def render: String = version match {
-    case MajorBinary(major) => s"scala $major"
-    case MinorBinary(major, minor) if major == 0 && minor < 30 =>
-      s"dotty $version"
-    case PreReleaseBinary(major, _, _, _) if major == 3 => s"scala $toString"
+    case MajorBinary(3) => s"scala 3"
+    case MinorBinary(0, minor) if minor < 30 => s"dotty $version"
+    case MinorBinary(3, _) => s"scala 3"
+    case PatchBinary(3, _, _) => s"scala 3"
```

Comment:
IIUC for all versions 3.x.y we will only have `_3` artifacts so it makes sense to consider all versions simply as Scala 3

Reply:
Here `version` is the binary version so it is always supposed to be `Major(3)` except for pre-release version of 3.0.0 (eg. `PreRelease(3, 0, Some(0), "RC1")`.

So `case MinorBinary(3, _)` and `case PatchBinary(3, _, _)` should never happen.

Also `case PreReleaseBinary(major, _,  _, _)` is the old dotty case (eg. `0.27.0-RC1`) and is is already filtered out by `Scala3Version.isValid` method, so it should never happen.

 ```suggestion
    case MajorBinary(3) => s"scala 3"
```

Reply:
Can we change the class constructor or put an assertion in it?
Having a `case _ => toString` looks like a hack not to have a exhaustivity check warning.
Maybe we could encode the fact that we are only expecting `Major(3)` into the types...

Reply:
I agree about the `case _ => toString` but the `case PreRelease(3, 0, Some(0), _)` is still valid.